kernel/timeout.c

```diff
@@ -77,11 +77,6 @@ static int32_t next_timeout(void)
 		ret = MAX(0, to->dticks - ticks_elapsed);
 	}
 
-#ifdef CONFIG_TIMESLICING
-	if (_current_cpu->slice_ticks && _current_cpu->slice_ticks < ret) {
-		ret = _current_cpu->slice_ticks;
-	}
-#endif
```

Comment:
Note how **all** the timeslice logic disappears from timeout.c now, with the sole exception of calling z_time_slice() out of the ISR.  Timeouts merely provide a generic API to the scheduler now, which is how is should have been.